### PR TITLE
Attempt to describe deadlock in use-when expressions

### LINKED ISSUE #1051: Deadlock conditions in `use-when` aren't discussed
Consider:

```xml
<p:declare-step type="cx:lib2d1" use-when="not(p:step-available('cx:lib2d2'))">
  <p:output port="result"/>
  <p:identity>
    <p:with-input><doc>lib1s1</doc></p:with-input>
  </p:identity>
</p:declare-step>

<p:declare-step type="cx:lib2d2" use-when="not(p:step-available('cx:lib2d1'))">
  <p:output port="result"/>
  <p:identity>
    <p:with-input><doc>lib1s1</doc></p:with-input>
  </p:identity>
</p:declare-step>
```

It's relatively difficult to create such a deadlock. You have to put the `use-when` expression on a `p:declare-step` and it has to include a call to `p:step-available`. There are a number of options for resolving them, but I think the simplest thing to do for 3.0 is make this a static error.

### PR DESCRIPTION
**Pull request by ndw**
Close #1051

### CODE REVIEW

File: xproc/src/main/xml/specification.xml [hunk 1]
```diff
@@ -3768,6 +3768,39 @@ linkend="initiating">static analysis</link> of the pipeline.</para>
...
+<para><error code="S0115">It is a <glossterm>static error</glossterm>
+if two or more elements are contained within a deadlocked network of
+<tag class="attribute">[p:]use-when</tag> expressions.</error> In
+order to avoid deadlock, there must exist an order in which every
+expression can be resolved without reference to an expression that
+occurs after it in the ordering.</para>
+
+<para>Processors may be required to evaluate expressions in an
+arbitrary order, but they are not required to solve a set of linear
+equations simultanously. So, while “declare both <code>ex:A</code> and
```
Comment (gimsieke) [inline xproc/src/main/xml/specification.xml:3797]: simultaneously

File: xproc/src/main/xml/specification.xml [hunk 2]
```diff
@@ -3768,6 +3768,39 @@ linkend="initiating">static analysis</link> of the pipeline.</para>
...
+<code>ex:B</code>” is a valid solution to the example above,
+conformation processors are not required (or allowed) to find it
```
Comment (gimsieke) [inline xproc/src/main/xml/specification.xml:3799]: conformant

File: xproc/src/main/xml/specification.xml [hunk 3]
```diff
@@ -3768,6 +3768,39 @@ linkend="initiating">static analysis</link> of the pipeline.</para>
...
+<code>ex:B</code> should be declared, and vice versa.</para>
+
+<para><error code="S0115">It is a <glossterm>static error</glossterm>
+if two or more elements are contained within a deadlocked network of
+<tag class="attribute">[p:]use-when</tag> expressions.</error> In
+order to avoid deadlock, there must exist an order in which every
+expression can be resolved without reference to an expression that
+occurs after it in the ordering.</para>
+
+<para>Processors may be required to evaluate expressions in an
```
Comment (gimsieke) [inline xproc/src/main/xml/specification.xml:3795]: may be required → may be allowed?

**Review by gimsieke — CHANGES_REQUESTED** [review]

File: xproc/src/main/xml/specification.xml (hunk 1 above)
Comment (ndw) [reply xproc/src/main/xml/specification.xml:3797]: 👍

File: xproc/src/main/xml/specification.xml (hunk 2 above)
Comment (ndw) [reply xproc/src/main/xml/specification.xml:3799]: 👍

File: xproc/src/main/xml/specification.xml (hunk 3 above)
Comment (ndw) [reply xproc/src/main/xml/specification.xml:3795]: I think required is correct. The processor has to determine an order that works, and whether or not there is one, and then evaluate the expressions in that order.

**Review by gimsieke — APPROVED** [review]

**Pull request merged**